Src/AutoMoq/MockType.cs

```diff
@@ -68,11 +68,14 @@ internal static Type GetMockedType(this Type type)
             where TMock : class
         {
             return setup.Returns(() =>
-                {
-                    var result = (TResult) context.Resolve(typeof (TResult));
-                    setup.Returns(result);
-                    return result;
-                });
+            {
+                var specimen = context.Resolve(typeof (TResult));
+                TResult result = specimen is TResult ? (TResult) specimen : default(TResult);

```

Comment:
I suppose we could alternatively throw an exception (.e.g, `ObjectCreationException`) to signal that the return value could not be instantiated.
This would happen when the context returns an `OmitSpecimen`, a `NoSpecimen`, or anything that is not of type `TResult`.